### PR TITLE
Donator Refactor and tweaks

### PR DESCRIPTION
**Pull request by Super3222**
- Code Refactor in preferences.dm to handle more than just donator hats.
- Split donor_start_items (list) into hats, outfits, suits, and shoes
- Changed the description of a few donator items (and also the captains hat)
- Microwave hat's now have a suicide act.
- Cowboy hats can now pull out candycorns like the detective's hat.
- Drinking hats are now fillable. You need to place in an unopened soda can. Comes with their own action button!
- Soda cans now start unopened. You need to activate them in your hand to crack them open, and once that happens you'll be able to drink out of it. Comes with a refreshing sound.

:cl: Super3222
rscadd: !!! DONOR FUN !!!
rscadd: Drinking hats can now be filled with unopened soda cans. Comes with an action button! Use it with the hat in your hands to empty the cans.
rscadd: Soda Cans need to cracked open to drink. Comes with their own sound!
rscadd: You can pull out candy corn from cowboy hats using alt+click!
rscadd: Microwave hat now has it's own suicide message!
rscadd: Most of the donor hats/outfits/suits have a new description!
rscadd: Donor items were split up into hats, outfits, suits, and shoes.
rscadd: Sith suit can be equipped in your jumpsuit slot, rather then your armor slot.
rscadd: !!! DONOR FUN END !!!
/:cl:

### CODE REVIEW

File: code/modules/clothing/head/jobs.dm [hunk 1]
```diff
@@ -23,7 +23,7 @@
 //Captain
 /obj/item/clothing/head/caphat
 	name = "captain's hat"
-	desc = "It's good being the king."
+	desc = "It's feels good to be king."
```
Comment (george99g) [inline code/modules/clothing/head/jobs.dm]: I'm going to be a grammar Nazi here and say that the `'s` is not needed there.

File: code/modules/food_and_drinks/drinks/drinks.dm [hunk 2]
```diff
@@ -272,6 +283,10 @@
 		var/obj/item/trash/can/crushed_can = new /obj/item/trash/can(user.loc)
 		crushed_can.icon_state = icon_state
 		qdel(src)
```
Comment (Cruix) [inline code/modules/food_and_drinks/drinks/drinks.dm:285]: You should have a special check for this block so you can't crush a can if it is in a hat, or at least remove the hat's reference to the can when you crush it.

Comment (ShadowDeath6) [inline code/modules/food_and_drinks/drinks/drinks.dm:285]: how would one go about crushing a can when it's inside a hat?

Comment (Super3222) [inline code/modules/food_and_drinks/drinks/drinks.dm:285]: The can is inside of the hat.

**Comment by Xantam-13** [comment]
Is the cueball helmet/suit back in? :(

**Comment by Super3222** [comment]
> Is the cueball helmet/suit back in? :(

Nah, it looks like it isn't. I think those are inside of costume vendor.

**Comment by X-TheDark** [comment]
Merge conflict caused by #604 - Changes to `preferences.dm` and `preferences_savefile.dm`

**Pull request closed without merging**